Automattic/simplenote-macos
# Add codefreeze automation

This PR adds a Fastlane function to automate some of the code freeze steps of the release process.

# To Test
### Setup
1. Run `bundle install`. Make sure you setup the env file as described in https://github.com/Automattic/simplenote-macos/pull/340.

### Test
1. Run `bundle exec fastlane code_freeze`.
2. When prompted, verify that the system reads the correct version and propose the right one. Confirm.
3. Verify that the operation finishes without errors.
4. Verify that a new `release/[next_release]` (currently 1.6) has been created, the branch protection set and that the version has been updated correctly in the xcconfig file.
5. Verify that the proper tag has been added (should be 1.6.0.0).
6. Verify that the proper milestone (currently 1.6) has been marked with the snowflake tag.
7. Verify that the a simplenotemacos_prs_list_xx_xx.txt file has been created in the home folder with the changelog.

### Clean up
8. Remove the `1.6.0.0` tag from your local repository and from GitHub.
9. Remove the snowflake tag from the milestone.
10. Delete the new release branch from your local repository and from GitHub.
